### DISCUSSION TITLE
Constraints with no Coefficients cause any issue?

### DISCUSSION THREAD

**Discussion by GhayourAli** [google/or-tools#3625] [Q&A] [answered]
I am using .Net wrapper of Google OR Tools to solve a Mixed Integer Programming problem using SCIP solver and have a couple of questions

1. I am dynamically generating constraints based on the data and at times it is possible that I create a constraint but then down the code no condition satisfy the need for that particular constraint. So if I create a constraint but don't set its coefficient, will it cause any issue or will that constraint be automatically ignored? I could not find an interface to remove a constraint from the solver.

2. I have set 8 threads for my solver and I am assuming that it runs different solver on each thread, is this assumption correct?

3. In case of infeasibilities, is there a way to identify which constraints or variables are causing the infeasibility so it can help better determine the root cause for infeasibility?

**Comment by lperron** [comment, marked as answer]
1) no impact
2) we disable parallelism on scip as (1) it brings very little performance improvement, and (2) it is buggy, it leaks memory...
3) This is called IIS, and is not accessible currently in OR-Tools.

**Reply by GhayourAli** [reply]
Thank you for the response. Is IIS planned to be made available in OR-Tools and any probable timeline for it?

**Comment by lperron** [comment]
Currently, not planned.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00



Le lun. 9 janv. 2023 à 11:13, GhayourAli ***@***.***> a
écrit :
